tests/SpecTests/RetryableReads/Prose2_RetryOnMongosTest.php

```diff
@@ -65,7 +64,8 @@ public function testRetryOnDifferentMongos(): void
 
         // Step 4: Enable failed command event monitoring for client
         $subscriber = new class implements CommandSubscriber {
-            public $commandFailedServers = [];
+            /** @var int[] */
+            public array $commandFailedServers = [];
```

Comment:
Later, this was used for the following assertion: 

```php
/* Step 6: Assert that two failed command events occurred. Assert that
 * the failed command events occurred on different mongoses. */
```

Per [`getServerConnectionId()`](https://www.php.net/manual/en/mongodb-driver-monitoring-commandstartedevent.getserverconnectionid.php), the connection ID is unique to each server and is only reported by MongoDB 4.2+. I think the correct change here would have been to replace `getServer()` with a concatenation of the server host/port. I'll make the change in a subsequent PR.

As-is, I think this might sometimes fail if both mongoses reported the same connection ID or if we test on mongos 4.0.